File: 백준/Silver/10828. 스택/스택.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Stack;
import java.util.StringTokenizer;

public class Main {

    static Stack<Integer> stack;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        stack = new Stack<>();

        for(int i=0; i<n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            switch (st.nextToken()) {
                case "push":
                    int num = Integer.parseInt(st.nextToken());
                    stack.push(num);
                    break;
                case "top":
                    if(!stack.isEmpty()) System.out.println(stack.peek());
                    else System.out.println(-1);
                    break;
                case "size":
                    System.out.println(stack.size());
                    break;
                case "pop":
                    if(!stack.isEmpty()) System.out.println(stack.pop());
                    else System.out.println(-1);
                    break;
                case "empty":
                    if(stack.isEmpty()) System.out.println(1);
                    else System.out.println(0);
                    break;
                default:
                    break;

            }
        }
    }
}

File: 백준/Silver/10828. 스택/스택.kt
import java.io.*

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val sb = StringBuilder()
    val stack = ArrayDeque<Int>()
    
    repeat(n) {
        val input = br.readLine().split(" ")
        when (input[0]) {
            "push" -> stack.addLast(input[1].toInt())
            "top" -> sb.appendLine(if (stack.isEmpty()) -1 else stack.last())
            "size" -> sb.appendLine(stack.size)
            "pop" -> sb.appendLine(if (stack.isEmpty()) -1 else stack.removeLast())
            "empty" -> sb.appendLine(if (stack.isEmpty()) 1 else 0)        
        }
    }
    
    print(sb)
}